### PR TITLE
Add route check, make step completions optional and non-fatal on error

### PR DESCRIPTION
**Pull request by joelsmith**
Any error on a step will be recorded but will not cause a run to
terminate. Only a timeout or a successful route check will cause
a run to terminate.

### CODE REVIEW

**Comment by joelsmith** [comment]
@twiest @ironcladlou PTAL. I'm still not convinced I have the right way to pass metrics around, so for now I'm keeping those changes out and just adding the route check and the focus on it (and de-emphasizing the other checks).

**Review by twiest — APPROVED** [review]
Looks great!

File: cmd/monitor/monitor.go [hunk 1]
```diff
@@ -161,7 +177,31 @@ func cleanupWorkspace(config *Config, clients client.RESTClients) error {
 	return nil
 }
 
-func newApp(config *Config, clients client.RESTClients) error {
+// pollURL will continually make HTTP GET requests of url until it returns status code 200 or times out
+// pollURL returns true if the request (eventually) succeeded with status code 200, and false if it timed out
+func pollURL(url string, timeoutTime time.Time) bool {
```
Comment (ironcladlou) [inline cmd/monitor/monitor.go:182]: k8s.io/apimachinery/pkg/util/wait.Until

File: cmd/monitor/monitor.go [hunk 2]
```diff
@@ -121,10 +134,13 @@ func main() {
...
 		}
-		if elapsed := time.Since(startTime); elapsed < interval {
-			fmt.Printf("Sleeping for %v before next iteration\n", interval-elapsed)
-			time.Sleep(interval - elapsed)
+		// sleep a minimum of 30 seconds between runs
+		toSleep := 30 * time.Second
+		if elapsed := time.Since(startTime); interval-elapsed > toSleep {
+			toSleep = interval - elapsed
 		}
+		fmt.Printf("Sleeping for %v before next iteration\n", toSleep)
```
Comment (ironcladlou) [inline cmd/monitor/monitor.go:142]: Please use glog to control log spam with verbosity levels

Comment (twiest) [reply cmd/monitor/monitor.go:142]: Yeah, this is on me. Joel said he'd start adding FIXME comments for lines like this. I'll replace those once I get to the glog stuff.

**Pull request merged**